src/components/app.tsx

```diff
@@ -106,8 +108,15 @@ export const App = () => {
 
   useAutoWidth({
     container: document.body,
-    onWidthChange: useCallback(
-      (newWidth) => { setGraphWidth(newWidth - (2 * SIDE_MARGIN_PLUS_BORDER)); }
+    onWidthChange: useCallback( (containerWidth) => {
+        const newWidth = containerWidth - (2 * SIDE_MARGIN_PLUS_BORDER);
+        setGraphWidth(newWidth);
+        const graphContainer = document.querySelector(".main-controls-and-waves-container");
+        if (graphContainer) {
+          const graphContainerHeight = graphContainer.clientHeight;
```

Comment:
I bet it could also be handled by ResizeObserver, but I'm assuming we care about height change only when the width changes.